src/iotconnect_telemetry.c

```diff
@@ -267,6 +362,6 @@ void iotcl_destroy_serialized(const char *serialized_string) {
 void iotcl_telemetry_destroy(IotclMessageHandle message) {
     if (message) {
         cJSON_Delete(message->root_value);
+        free(message);
     }
-    free(message);
```

Comment:
this could potentially be a NULL pointer, so moving this inside the check prevents that

Reply:
You can free NULL -- so it's probably not a big improvement.